yb-voyager/cmd/exportDataDebezium.go

```diff
@@ -185,19 +194,61 @@ func prepareDebeziumConfig(tableList []*sqlname.SourceName, tablesColumnList map
 	return config, tableNameToApproxRowCountMap, nil
 }
 
-// required only for postgresql since GetAllTables() returns all tables and partitions
-func filterTablePartitions(tableList []*sqlname.SourceName) []*sqlname.SourceName {
-	if source.DBType != POSTGRESQL || source.TableList != "" {
-		return tableList
+// required only for postgresql/yugabytedb since GetAllTables() returns all tables and partitions
+func addLeafPartitionsInTableList(tableList []*sqlname.SourceName) ([]*sqlname.SourceName, error) {
+	requiredForSource := source.DBType == "postgresql" || source.DBType == "yugabytedb"
+	if !requiredForSource  {
+		return tableList, nil
 	}
 
-	filteredTableList := []*sqlname.SourceName{}
+	modifiedTableList := []*sqlname.SourceName{}
+	//TODO: test when we upgrade to PG13+ as partitions are handled with root table 
+	//Refer- https://debezium.zulipchat.com/#narrow/stream/302529-community-general/topic/Connector.20not.20working.20with.20partitions
 	for _, table := range tableList {
-		if !source.DB().IsTablePartition(table) {
-			filteredTableList = append(filteredTableList, table)
+		rootTable, err := GetRootTableOfPartition(table)
+		if err != nil {
+			return nil, fmt.Errorf("failed to get root table of partition %s: %v", table.Qualified.MinQuoted, err)
+		}
+		allLeafPartitions := GetAllLeafPartitions(table)
+		if len(allLeafPartitions) == 0 {
+			if rootTable != table { //child table is present in list through table-list flag
+				PartitionsToRootTableMap[table.Qualified.MinQuoted] = rootTable.Qualified.MinQuoted
+			}
+			modifiedTableList = append(modifiedTableList, table)
+		} else {
+			for _, childPartition := range allLeafPartitions {
+				PartitionsToRootTableMap[childPartition.Qualified.MinQuoted] = rootTable.Qualified.MinQuoted
+			}
+			modifiedTableList = append(modifiedTableList, allLeafPartitions...)
+		}
+	}
+	return lo.Uniq(modifiedTableList), nil
+}
+
+func GetRootTableOfPartition(table *sqlname.SourceName) (*sqlname.SourceName, error){
+	parentTable := source.DB().ParentTableOfPartition(table)
+	if parentTable == "" {
+		return table, nil
+	}
+	defaultSourceSchema, noDefaultSchema := getDefaultSourceSchemaName()
+	if noDefaultSchema {
+		return nil, fmt.Errorf("default schema not found")
+	}
```

Comment:
Use `table.SchemaName.Unquoted` as the "default schema name" to pass to the `NewSourceNameFromMaybeQualifiedName()`.

Reply:
Actually, no can't use the schema of the table as the default schema because if the partition is non-public the result from the db query will include a qualified name always and only in the case of public schema it will not be qualified. So, we have to use default only and partitions can be cross schemas as well so we can't consider a table's schema to be the same as its root table or vice versa.